### PR TITLE
Fix bottom buttons overflow when button content is too long

### LINKED ISSUE #2658: Chrome in french: display glitch
### Description
Screenshot from 2024-02-28 14-16-22

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
https://github.com/linagora/tmail-flutter/issues/2658

### Screenshots

**Long text button with overflow**
Screenshot 2024-03-11 at 14 51 14

**Long text button**
Screenshot 2024-03-11 at 14 51 50

**Normal length button**
Screenshot 2024-03-11 at 14 52 18

**Mobile long text**
Screenshot 2024-03-12 at 14 40 45

### Responsive Demo

https://github.com/linagora/flutter-date-range-picker/assets/160106668/735ddb14-0daf-4c79-b024-b41b082074b6

### CODE REVIEW

**Comment by dab246** [comment]
Please demo video for responsive web

File: lib/multiple_view_date_range_picker.dart [hunk 1]
```diff
@@ -66,6 +66,11 @@ class MultipleViewDateRangePicker extends StatefulWidget {
 
 class _MultipleViewDateRangePickerState extends State<MultipleViewDateRangePicker> {
   final String dateTimePattern = 'dd/MM/yyyy';
+  final desktopDateRangePickerWidth = 650.0;
+  final tabletDateRangePickerBottomPadding = 16.0;
```
Comment (dab246) [inline lib/multiple_view_date_range_picker.dart]: Please use `const`

File: lib/multiple_view_date_range_picker.dart [hunk 2]
```diff
@@ -66,6 +66,11 @@ class MultipleViewDateRangePicker extends StatefulWidget {
...
+  final startAndEndDateInputTabletKey = GlobalKey();
+  final bottomViewTabletConfirmationButtonsKey = GlobalKey();
+  bool bottomViewTabletConfirmationButtonsOverflow = false;
```
Comment (dab246) [inline lib/multiple_view_date_range_picker.dart]: please use private variable

File: lib/multiple_view_date_range_picker.dart [hunk 3]
```diff
@@ -88,6 +93,30 @@ class _MultipleViewDateRangePickerState extends State<MultipleViewDateRangePicke
...
     super.initState();
+    _checkIfTabletBottomButtonIsOverflowed();
+  }
+
+  void _checkIfTabletBottomButtonIsOverflowed() {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final startAndEndDateInputTabletRenderBox = startAndEndDateInputTabletKey
+        .currentContext?.findRenderObject() as RenderBox?;
+      final bottomViewTabletConfirmationButtonsRenderBox = bottomViewTabletConfirmationButtonsKey
+        .currentContext?.findRenderObject() as RenderBox ?;
```
Comment (dab246) [inline lib/multiple_view_date_range_picker.dart]:
```suggestion
        .currentContext?.findRenderObject() as RenderBox?;
```

File: lib/multiple_view_date_range_picker.dart [hunk 4]
```diff
@@ -433,44 +462,26 @@ class _MultipleViewDateRangePickerState extends State<MultipleViewDateRangePicke
           ],
         ),
       );
-    } else {
+    } else if (!bottomViewTabletConfirmationButtonsOverflow) {
       return Padding(
-        padding: const EdgeInsets.all(16),
+        padding: EdgeInsets.all(tabletDateRangePickerBottomPadding),
```
Comment (dab246) [inline lib/multiple_view_date_range_picker.dart]: Please use `const` for widget if possible

File: lib/multiple_view_date_range_picker.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: Screenshot 2024-03-12 at 10 56 58

File: lib/multiple_view_date_range_picker.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: variables inside `State` object cannot be `const`, only `final`. Hence, `const` cannot be applied here.

File: lib/multiple_view_date_range_picker.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: Done

File: lib/multiple_view_date_range_picker.dart (hunk 3 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: Done

File: lib/multiple_view_date_range_picker.dart (hunk 1 above)
Comment (dab246) [reply lib/multiple_view_date_range_picker.dart]: That is correct. Use `const` as much as possible.

File: lib/multiple_view_date_range_picker.dart (hunk 4 above)
Comment (dab246) [reply lib/multiple_view_date_range_picker.dart]: So let's give it a `const` value. If you don't do this, the widget tree may be re-rendered when the state changes.

File: lib/multiple_view_date_range_picker.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: Done

File: lib/multiple_view_date_range_picker.dart (hunk 4 above)
Comment (tddang-linagora) [reply lib/multiple_view_date_range_picker.dart]: Done

**Review by dab246 — APPROVED** [review]
Please rebase it

**Pull request merged**